src/fiatconnect/index.ts

```diff
@@ -60,6 +65,40 @@ export async function getFiatConnectProviders(
   return providers
 }
 
+/**
+ * Logs in with a FiatConnect provider. Will not attempt to log in if an
+ * unexpired session already exists, unless the `forceLogin` flag is set to `true`.
+ * If the user's wallet is currently locked, will prompt for PIN entry.
+ */
+export async function loginWithFiatConnectProvider(
+  wallet: UnlockableWallet,
+  fiatConnectClient: FiatConnectApiClient,
+  forceLogin: boolean = false
+): Promise<void> {
+  if (fiatConnectClient.isLoggedIn() && !forceLogin) {
+    return
+  }
+
+  const [account] = wallet.getAccounts()
+  if (!wallet.isAccountUnlocked(account)) {
+    await wallet.unlockAccount(account, await getPassword(account), UNLOCK_DURATION)
+  }
+
+  const response = await fiatConnectClient.login()
+  if (!response.isOk) {
+    Logger.error(TAG, `Failure logging in with FiatConnect provider: ${response.error}, throwing`)
+    throw response.error
+  }
+}
+
+export function getSigningFunction(wallet: UnlockableWallet): (message: string) => Promise<string> {
```

Comment:
what will happen if the FiatConnectApiClient attempts to use this after the wallet has locked? Should this function also deal with that scenario (either by attempting to unlock, or throwing an error that we can catch somewhere)?